pkg/generator/password_reconciler.go

```diff
@@ -166,30 +166,33 @@ func (r *PasswordReconciler) generate(password *sgv1alpha1.Password) (string, er
 
 	remainingLength := spec.Length - spec.Symbols - spec.Digits - spec.UppercaseLetters - spec.LowercaseLetters
 	for i := 0; i < remainingLength; i++ {
-		value, err := randomElement(allCharSet)
+		value, err := r.randomElement(allCharSet)
 		if err != nil {
 			return "", err
 		}
 		generatedPassword.WriteString(value)
 	}
 
 	inRune := []rune(generatedPassword.String())
-	localCryptoShuffle(len(inRune), func(i, j int) {
+	err := r.localCryptoShuffle(len(inRune), func(i, j int) {
 		inRune[i], inRune[j] = inRune[j], inRune[i]
 	})
+	if err != nil {
+		return "", err
+	}
 
 	return string(inRune), nil
 }
 
-func randomElement(s string) (string, error) {
+func (*PasswordReconciler) randomElement(s string) (string, error) {
 	n, err := rand.Int(rand.Reader, big.NewInt(int64(len(s))))
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("Getting rand int: %s", err)
 	}
 	return string(s[n.Int64()]), nil
 }
 
-func localCryptoShuffle(n int, swap func(i, j int)) {
+func (*PasswordReconciler) localCryptoShuffle(n int, swap func(i, j int)) error {
 	if n < 0 {
 		panic("invalid argument to Shuffle")
```

Comment:
This should not panic the generator unless it is an internal inconsistency error. This looks like invalid input from the spec itself, since we now have an output error maybe we can return it instead.